pkg/kv/kvserver/batcheval/cmd_refresh_range.go

```diff
@@ -81,15 +83,24 @@ func refreshRange(
 	defer iter.Close()
 
 	var meta enginepb.MVCCMetadata
-	iter.SeekGE(storage.MakeMVCCMetadataKey(span.Key))
-	for {
+	for iter.SeekGE(storage.MVCCKey{Key: span.Key}); ; iter.Next() {
 		if ok, err := iter.Valid(); err != nil {
 			return err
 		} else if !ok {
 			break
 		}
 
 		key := iter.UnsafeKey().Clone()
+
+		if _, hasRange := iter.HasPointAndRange(); hasRange {
+			rangeKVs := iter.RangeKeys()
+			if len(rangeKVs) == 0 { // defensive
```

Comment:
Do we need to be defensive here because of the time bounds that could filter all range keys while still reporting has range?